escrow/payment_handler.go

```diff
@@ -93,7 +96,39 @@ func (h *paymentChannelPaymentHandler) getPaymentFromContext(context *handler.Gr
 }
 
 func (h *paymentChannelPaymentHandler) Complete(payment handler.Payment) (err *handler.GrpcError) {
-	return paymentErrorToGrpcError(payment.(*paymentTransaction).Commit())
+	if err = paymentErrorToGrpcError(payment.(*paymentTransaction).Commit()); err != nil {
+		PublishChannelStats(payment)
+	}
+	return err
+}
+
+func PublishChannelStats(payment handler.Payment) (err *handler.GrpcError) {
+	if !config.GetBool(config.MeteringEnabled)  {
+		err = handler.NewGrpcErrorf(codes.Internal, "Cannot post latest offline channel state as metering is disabled !!")
+		log.WithError(err.Err())
+		return err
+	}
+	paymentTransaction := payment.(*paymentTransaction)
+	channelStats := &metrics.ChannelStats{ChannelId: paymentTransaction.payment.ChannelID,
+		AuthorizedAmount:paymentTransaction.payment.Amount,
+		FullAmount:paymentTransaction.Channel().FullAmount,
+		Nonce: paymentTransaction.Channel().Nonce,
+		GroupID:blockchain.BytesToBase64(paymentTransaction.Channel().GroupID[:]),
+	}
+	serviceURL := config.GetString(config.MeteringEndPoint)+"/contract-api/channel/"+channelStats.ChannelId.String()+"/consume"
+
+	channelStats.OrganizationID = config.GetString(config.OrganizationId)
+	channelStats.ServiceID = config.GetString(config.ServiceId)
+	log.Debug(channelStats)
```

Comment:
I would add some descibing message:
```suggestion
	log.Debugf("Payment handler is publishing channel statistics: %v", channelStats)
```